File: Javascript/2938-separate-black-and-white-balls.js
/**
 * @param {string} s
 * @return {number}
 */
var minimumSteps = function(s) {
    let left = 0, right = s.length - 1, swaps = 0;
    while (left < right)
    {
        if (s[left] === '0')
        {
            left++;
        }
        else
        {
            if (s[right] === '0')
            {
                swaps += right - left;
                left++;
            }
            right--;
        }
    }
    return swaps;
};

File: Java/2938-separate-black-and-white-balls.java
class Solution {
    public long minimumSteps(String s) {
        int left = 0, right = s.length() - 1;
        long swaps = 0;

        while (left < right) {
            char chL = s.charAt(left);
            if (chL == '0') {
                left++;
            }
            else {
                char chR = s.charAt(right);
                if (chR == '0') {
                    swaps += right - left;
                    left++;
                }
                right--;
            }
        }

        return swaps;
    }
}